src/main/java/com/xceptance/xlt/engine/DataManagerImpl.java

```diff
@@ -390,4 +394,22 @@ static StringBuilder removeLineSeparators(final StringBuilder src, final char re
 
         return src;
     }
+    
+    /**
+     * {@inheritDoc}
+     */
+    public DataLogger dataLogger(String scope)
+    {
+        DataLogger dataLogger = dataLoggers.get(scope);
+        
+        // check if dataLogger has already been initialized
+        // otherwise create one, add it to the list and return it
+        if (dataLogger == null)
+        {            
+            dataLogger = new DataLoggerImpl(session, scope);
+            dataLoggers.put(scope, dataLogger);
+        }
+        
+        return dataLogger;
```

Comment:
```suggestion
        return dataLoggers.computeIfAbsent(scope, sc -> new DataLoggerImpl(session, sc));
```

This is shorter and also performs some clever magic to make adding a value thread-safe because the underlying map is a `ConcurrentHashMap`.